python/python/res/simulator/batch_simulator.py

```diff
@@ -50,27 +50,48 @@ def __init__(self, res_config, controls, results):
             raise ValueError("The first argument must be valid ResConfig instance")
 
         self.res_config = res_config
-        self.ert = EnKFMain( self.res_config )
-        self.control_keys = []
-        self.result_keys = []
+        self.ert = EnKFMain(self.res_config)
+        self.control_keys = tuple(controls.keys())
+        self.result_keys = tuple(results)
 
         ens_config = self.res_config.ensemble_config
-        for key in controls.keys():
-            ens_config.addNode( EnkfConfigNode.create_ext_param( key , controls[key] ))
-            self.control_keys.append(key)
+        for control_name, variable_names in controls.iteritems():
+            ens_config.addNode(EnkfConfigNode.create_ext_param(control_name, variable_names))
 
         for key in results:
-            ens_config.addNode( EnkfConfigNode.create_gen_data( key , "{}_%d".format(key)))
-            self.result_keys.append(key)
+            ens_config.addNode(EnkfConfigNode.create_gen_data(key, "{}_%d".format(key)))
 
 
+    def _setup_case(self, case, file_system):
+        for sim_id, (geo_id, controls)  in enumerate(case):
+            assert isinstance(geo_id, int)
+
+            node_id = NodeId(0, sim_id)
+            if set(controls.keys()) != set(self.control_keys):
+                err_msg = "Mismatch between initialized and provided control names."
+                raise KeyError(err_msg)
+
+            for control_name, control in controls.iteritems():
+                ens_config = self.res_config.ensemble_config
```

Comment:
I would prefer if the `ens_config` variable  was moved out of both the loops.

Reply:
It is just a shorter name for the `ensemble_config` in question. In my opinion it is odd to define it before needed. How about skipping the shorter name as it is only used once and go for
```python
node = EnkfNode(self.res_config.ensemble_config[control_name])
```
78 characters long and good to go?